LiskHQ/lisk-core
# Discontinue the chain after the snapshot height finalizes

### What was the problem?

This PR resolves #924 

### How was it solved?

- [x] Implement the ShutdownNodePlugin
- [x] Bump Lisk Core version
- [x] Set `system.backup.height` and update `rpc.allowedMethods` for the default mainnet config

### How was it tested?

- Build the app: `yarn build`
- Start node with the following `custom-config.json`: `./bin/run start -n mainnet -c ./custom-config.json`
```json
{
	"system": {
		"backup": {
			"height": 24694375
		}
	}
}
```
- Expected result:
```sh
2024-05-06T13:37:30.694Z FATAL Sameers-MBP.koepi126 application 48715 [err=Triggering node shutdown from Shutdown plugin: current height: 24694765; reason: finalized height 24694615 is greater than the given shutdown height 24694478 (snapshot height: 24694375 + buffer: 103)] System error: unhandledRejection
2024-05-06T13:37:30.694Z INFO Sameers-MBP.koepi126 application 48715 [errorCode=1 message=Triggering node shutdown from Shutdown plugin: current height: 24694765; reason: finalized height 24694615 is greater than the given shutdown height 24694478 (snapshot height: 24694375 + buffer: 103)] Application shutdown started
2024-05-06T13:37:30.694Z INFO Sameers-MBP.koepi126 engine 48715 Engine cleanup started
2024-05-06T13:37:30.694Z INFO Sameers-MBP.koepi126 engine 48715 Network cleanup started
2024-05-06T13:37:30.697Z ERROR Sameers-MBP.koepi126 engine 48715 [peerId=162.19.248.21:8001 err=Event 'rpc-request' was aborted due to a bad connection] Fail to handle received block
2024-05-06T13:37:30.697Z INFO Sameers-MBP.koepi126 engine 48715 Network cleanup completed
2024-05-06T13:37:30.700Z INFO Sameers-MBP.koepi126 engine 48715 Engine cleanup completed
2024-05-06T13:37:30.700Z INFO Sameers-MBP.koepi126 application 48715 Stopping Controller
2024-05-06T13:37:30.700Z INFO Sameers-MBP.koepi126 application 48715 Controller cleanup completed
2024-05-06T13:37:30.702Z INFO Sameers-MBP.koepi126 application 48715 [errorCode=1 message=Triggering node shutdown from Shutdown plugin: current height: 24694765; reason: finalized height 24694615 is greater than the given shutdown height 24694478 (snapshot height: 24694375 + buffer: 103)] Application shutdown completed
sameer@Sameers-MBP lisk-core % echo $?                                                                        
1
```
